### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by mxxt1**
# Description

Will complete this later this week. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: developer/developer-model.js [hunk 1]
```diff
@@ -8,17 +9,24 @@ module.exports = {
...
   let { startDate, endDate, count } = query
 
   let entries
   let totalCount
+  const shelfData = bookshelf.model('shelfData', {
+    tableName: 'platform_market_prices2'
+  })
+
+  // * WORKING EXAMPLE OF FETCHING ALL DATA FROM TABLE AND CONVERTING DATA.
+  console.log('BOOKSHELF DATA PULL', await new shelfData().fetchAll().then(res => res.toJSON()))
```
Comment (blackmacaroon) [inline developer/developer-model.js:29]: do we need this console log deployed?

File: developer/developer-model.js [hunk 2]
```diff
@@ -29,6 +37,7 @@ async function getSautiData(query,apiCount) {
     const nextDate = new Date(cursorArray[0])
     const nextId = cursorArray[1]
     let queryOperation = DBSt('platform_market_prices2')
+    // console.log(await typeof queryOperation)
```
Comment (blackmacaroon) [inline developer/developer-model.js:40]: log

File: developer/developer-model.js [hunk 3]
```diff
@@ -83,17 +92,19 @@ async function getSautiData(query,apiCount) {
       'date',
       'udate'
     )
-
+    // console.log(await queryOperation)
```
Comment (blackmacaroon) [inline developer/developer-model.js:95]: log

File: developer/developer-model.js [hunk 4]
```diff
@@ -83,17 +92,19 @@ async function getSautiData(query,apiCount) {
...
     if (startDate && endDate) {
       queryOperation = queryOperation.andWhereBetween('date', [
         startDate,
         endDate
       ])
     }
 
+    console.log('DATES', startDate, endDate)
```
Comment (blackmacaroon) [inline developer/developer-model.js:103]: log

File: developer/developer-model.js [hunk 5]
```diff
@@ -106,6 +117,9 @@ async function getSautiData(query,apiCount) {
   } else {
     // If user wants data from specific country/countries
     let queryOperation = DBSt('platform_market_prices2')
+
+    console.log((await queryOperation).length)
```
Comment (blackmacaroon) [inline developer/developer-model.js:121]: log

File: developer/developer-model.js [hunk 6]
```diff
@@ -165,31 +179,114 @@ async function getSautiData(query,apiCount) {
...
     totalCount = await queryOperation.clone().count()
+
+    //first set of records based on count, +1 record for next reference.
     entries = await queryOperation
       .where('active', (query.a = 1))
       .orderBy('date', 'desc')
       .orderBy('id', 'desc')
       .limit(Number(count) + 1)
+
+    // TODO: COMPLETE MANUAL FUNCTION FOR ORGANIZING A COMPLETE SET OF TRAVERSING DATA OBEJCT.
```
Comment (blackmacaroon) [inline developer/developer-model.js:202]: don't forget to clear Todo's

**Review by blackmacaroon — CHANGES_REQUESTED** [review]

**Comment by mxxt1** [comment]
This branch is outmoded. Work was done on the express-pagination branch.

**Pull request closed without merging**